datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -1458,11 +1552,15 @@ mod tests {
     // ----- Simplifier tests -------
     // ------------------------------
 
-    fn simplify(expr: Expr) -> Expr {
+    fn simplify_result(expr: Expr) -> Result<Expr> {
         let schema = expr_test_schema();
         let execution_props = ExecutionProps::new();
         let info = SimplifyContext::new(vec![&schema], &execution_props);
-        expr.simplify(&info).unwrap()
+        expr.simplify(&info)
+    }
+
+    fn simplify(expr: Expr) -> Expr {
```

Comment:
Why do we need this function?

Reply:
The new function `simplify_result` is used in the test `test_simplify_modulo_by_zero_non_null`

Reply:
Hmm, I see. 
You can use the original `simplify` function and use ``should_panic` to check the error message of `test_simplify_modulo_by_zero_non_null`.

Reply:
`should_panic` is more generic and can occur anywhere in the function. By checking `Result` for error we can specifically look for error in that function.

Reply:
You can use `[should_panic(expected = "Your error message")]` to precisely check where error occurs

Reply:
Cool! How do we know what's the error message for `Err(DataFusionError::ArrowError(ArrowError::DivideByZero))` ?

Reply:
The most straight way is trying on the `datafusion-cli` and find what error message is given

Reply:
Removed new function and using `should_panic`